pkg/option/config.go

```diff
@@ -3072,6 +3075,63 @@ func (c *DaemonConfig) KubeProxyReplacementFullyEnabled() bool {
 		c.EnableSessionAffinity
 }
 
+// StoreInFile stores the configuration in a the given directory under the file
+// name 'daemon-config.json'. If this file already exists, it is renamed to
+// 'daemon-config-1.json', if 'daemon-config-1.json' also exists,
+// 'daemon-config-1.json' is renamed to 'daemon-config-2.json'
+func (c *DaemonConfig) StoreInFile(dir string) error {
+	backupFileNames := []string{
+		"daemon-config.json",
+		"daemon-config-1.json",
+		"daemon-config-2.json",
+	}
+	backupFiles(dir, backupFileNames)
+	f, err := os.Create(backupFileNames[0])
+	if err != nil {
+		return err
+	}
+	e := json.NewEncoder(f)
+	e.SetIndent("", " ")
+	return e.Encode(c)
+}
+
+// StoreViperInFile stores viper's configuration in a the given directory under
+// the file name 'viper-config.yaml'. If this file already exists, it is renamed
+// to 'viper-config-1.yaml', if 'viper-config-1.yaml' also exists,
+// 'viper-config-1.yaml' is renamed to 'viper-config-2.yaml'
+func StoreViperInFile(dir string) error {
+	backupFileNames := []string{
+		"viper-config.yaml",
+		"viper-config-1.yaml",
+		"viper-config-2.yaml",
```

Comment:
Since this is probably what Cilium was configured with, maybe we can have a name that's clearer than this to indicate that?

Reply:
Also, should viper-config*.yaml rather be stored by date, so that we can diff on a timeline when agent was restarted with different configs?

Reply:
It's easier to leave without timestamps in the file names, however, the content of `agent-config.json` contains an entry with the creation time.